scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -127,7 +133,195 @@ def setUp(self):
+        self.admin.assert_icommand('ibun -x {} .'.format(self.bundled_coll))
+        test_coll = self.test_admin_coll_path
+        rule_file = ""
+        output=""
+        rule_header = dedent("""\
+            from genquery import *
+            def main(rule_args,callback,rei):
+                TestCollection = "{test_coll}"
+            """)
+        rule_footer = dedent("""\
+            INPUT null
+            OUTPUT ruleExecOut
+            """)
+
+        frame_rule = lambda text,indent=4: \
+            rule_header+('\n'+' '*indent+('\n'+' '*indent).join(dedent(text).split("\n"))
+                        )+"\n"+rule_footer
+        #--------------------------------
+        test_cases = [(
+            frame_rule('''\
+                L = []
+                for j in {octals}:
+                    cond_string= "COLL_NAME = '{{c}}' and DATA_NAME < '{{ltparam}}'".format(c=TestCollection,ltparam=j)
+                    q = Query(callback,
+                              columns=["DATA_NAME"],
+                              conditions=cond_string)
+                    L.append(q.total_rows())
+                callback.writeLine("stdout", repr(L))
+                '''),
+            "total_rows_",
+            lambda : eval(output) == interesting
+        ), ( #----------------
+            frame_rule('''\
+                row_format = lambda row:"{{COLL_NAME}}/{{order(DATA_NAME)}}".format(**row)
+                q = Query(callback,
+                          columns=["order(DATA_NAME)","COLL_NAME"],
+                          output=AS_DICT,
+                          offset=0,limit=1,
+                          conditions="COLL_NAME = '{{c}}'".format(c=TestCollection))
+                res_1=[row_format(r) for r in q]
+                row_count = q.total_rows()
+                res_2=[row_format(r) for r in q.copy(offset=row_count-1)]
+                callback.writeLine('stdout',repr( [len(x) for x in res_1,res_2] +\
+                                                  [
+                                                   (res_1 + res_2 ==
+                                                    [TestCollection+"/"+x for x in ["%04o"%y for y in 0,{max_count}]])
+                                                  ] ))
+                '''),
+            "offset_limit_",
+            lambda: True # lambda : output.replace(" ","").strip() == "[1,1,True]"
+        ), ( #----------------
+            frame_rule('''\
+                q = Query(callback,
+                          columns=["DATA_ID"],
+                          conditions="COLL_NAME = '{{c}}'".format(c=TestCollection))
+                q_iter = iter( q )
+                first_row = next(q_iter)
+                callback.writeLine('stdout',repr([1+len([row for row in q_iter]),
+                                                  q.copy().total_rows()]))
+                '''),
+            "copy_query_",
+            lambda : eval(output) == [ max_count+1 ]*2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of possibly insecure function - consider using safer ast.literal_eval.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5016322)